### PR TITLE
go/oasis-node/cmd: Allow using non local gRPC connections

### PR DESCRIPTION
**Pull request by tjanez**
Require server-side TLS for non unix-socket gRPC connections.

### CODE REVIEW

File: go/oasis-test-runner/scenario/e2e/stake_cli.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ const (
 
 	// Testing escrow account public key (hex-encoded).
 	escrowPubkeyHex = "6ea5328f943ef6f66daaed74cb0e99c3b1c45f76307b425003dbc7cb3638ed35"
+
+	// Oasis Protocol Foundation's public gRPC endpoint.
+	publicGrpcAddr = "grpc.oasis.dev:443"
```
Comment (kostko) [inline go/oasis-test-runner/scenario/e2e/stake_cli.go:68]: Should we disable this test by default? Otherwise this will make CI fail if the public endpoint is unavailable.

Comment (Yawning) [reply go/oasis-test-runner/scenario/e2e/stake_cli.go:68]: oasis-core intentionally is network instance neutral.  I'm not sure this feature warrants changing that policy...

Comment (kostko) [reply go/oasis-test-runner/scenario/e2e/stake_cli.go:68]: This is a test, not some default.

**Comment by Yawning** [comment]
Ultimately I don't see the point of this.  The only time when this really is an issue is when voting on governance, and the current voting period which ends in a day should be the last governance vote where this tooling needs to be used in the first place.

tldr; use the sdk-cli.  If the sdk-cli doesn't provide what you need, go add to it.

File: go/oasis-test-runner/scenario/e2e/stake_cli.go (hunk 1 above)
Comment (tjanez) [reply go/oasis-test-runner/scenario/e2e/stake_cli.go]: I went ahead and removed the test since it would complicate things and is not really that crucial.

**Comment by tjanez** [comment]
> Ultimately I don't see the point of this. The only time when this really is an issue is when voting on governance, and the current voting period which ends in a day should be the last governance vote where this tooling needs to be used in the first place.

I agree that the future of CLI tooling is to migrate to the new Oasis CLI we are developing in the Oasis SDK repo but I think this small change will still improve the UX until people migrate to that.

Another example where this would have impact is that people who don't have access to a synced node would like to use State Sync and [need the list of nodes that offer that](https://docs.oasis.dev/general/run-a-node/advanced/sync-node-using-state-sync#obtaining-addresses-of-oasis-nodes-publicly-exposed-grpc-endpoints).

With this change, they could use:
```
oasis-node registry node list -v -a grpc.oasis.dev:443 | \
  jq 'select(.roles | contains("consensus-rpc")) | .tls.addresses' 
 ```

**Review by kostko — APPROVED** [review]
